### DISCUSSION TITLE
Is there a way to render Markdown in theme extension files?

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3359] [Q&A]
Currently extending a theme to make a separate page I can use to insert and render specific files.

The main issue I encounter is, that Content in markdown won't be rendered to HTML due to the file being HTML too.

I was now wondering, if there is a way to render the markdown content in a theme extension file, so that I can display the content properly... I would like to keep the formatting to markdown itself.

**Comment by pawamoy** [comment]
I'd say it is possible if you have access to the Jinja environment. In that case you could:

1. somehow get MkDocs `config["markdown_extensions"], config["mdx_configs"]`
2. use them to instantiate a `Markdown` instance
3. add a filter to the Jinja env that uses it `convert` method
4. use the filter in your templates: `{% filter convert %}**Markdown contents**{% endfilter %}`

**Reply by Andre601** [reply]
Idk if I do...
I simply use theme extension here to make some custom pages, so idk if I have access to the jinja env here...
Also, not sure how this actually would look like if I do...

**Reply by pawamoy** [reply]
Not sure how themes are developed, but you can always combine it to a plugin that uses `on_env` to add your Jinja filters.

**Reply by Andre601** [reply]
I still don't really get it tbh...

Preferrably don't want to have a plugin just for this alone...

Tho, I assume the main issue here is, that YAML frontmatter is (obviously) not parsed to HTML, since what I assume is that themes receive the already rendered page content through the `{{ page.content }}` variable which is why this for them (I hope my text makes sense).
